creator-node/src/routes/nodeSync.js

```diff
@@ -240,165 +271,157 @@ async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint, dbOnlySync
+        throw new Error(`Cannot sync - imported data is not contiguous. Local max clock val = ${localMaxClockVal} and imported min clock val ${fetchedClockRecords[0].clock}`)
+      }
+
+      // All DB updates must happen in single atomic tx - partial state updates will lead to data loss
       const transaction = await models.sequelize.transaction()
 
+      /**
+       * Process all DB updates for cnodeUser
+       */
       try {
-        const cnodeUser = await models.CNodeUser.findOne({
+        req.logger.info(redisKey, `beginning add ops for cnodeUser wallet ${fetchedWalletPublicKey}`)
+
+        /**
+         * Update CNodeUser entry if exists else create new
+         *
+         * Cannot use upsert since it fails to use default value for cnodeUserUUID per this issue https://github.com/sequelize/sequelize/issues/3247
+         */
+
+        let cnodeUser
+
+        // Fetch current cnodeUser from DB
+        const cnodeUserRecord = await models.CNodeUser.findOne({
           where: { walletPublicKey: fetchedWalletPublicKey },
           transaction
         })
-        const fetchedLatestBlockNumber = fetchedCNodeUser.latestBlockNumber
-        const fetchedLatestClockVal = fetchedCNodeUser.clock
-
-        // Delete any previously stored data for cnodeUser in reverse table dependency order (cannot be parallelized).
-        if (cnodeUser) {
-          // Ensure imported data has higher blocknumber than already stored.
-          const latestBlockNumber = cnodeUser.latestBlockNumber
-          const latestClockValue = cnodeUser.clock
-
-          if (latestClockValue > fetchedLatestClockVal) {
-            throw new Error(`Imported data is outdated, will not sync. Imported latestBlockNumber \
-              ${fetchedLatestBlockNumber} Self latestBlockNumber ${latestBlockNumber}. \
-              fetched latestClockVal: ${fetchedLatestClockVal}, self latestClockVal: ${latestClockValue}`)
-          } else if (latestClockValue === fetchedLatestClockVal) {
-            // Already to update, no sync necessary
-            req.logger.info(redisKey, `User ${fetchedWalletPublicKey} already up to date! fetchedLatestClockVal=${fetchedLatestClockVal}, latestClockValue=${latestClockValue}`)
-            // the transaction declared outside the try/catch needs to be closed. if we call the continue
-            // and do not end the tx, it will never be closed
-            transaction.rollback()
-            continue
-          }
 
-          const cnodeUserUUID = cnodeUser.cnodeUserUUID
-          req.logger.info(redisKey, `beginning delete ops for cnodeUserUUID ${cnodeUserUUID}`)
-
-          const numAudiusUsersDeleted = await models.AudiusUser.destroy({
-            where: { cnodeUserUUID },
-            transaction
-          })
-          req.logger.info(redisKey, `numAudiusUsersDeleted ${numAudiusUsersDeleted}`)
+        // Update existing entry if found, else create new
```

Comment:
Does this occur on the first sync to a secondary?

Reply:
yeah existing entry is for all subsequent syncs, initial sync will enter the else case where no local cnodeUserRecord exists and create a new one. added comment to make this more explicit